src/Set.ts

```diff
@@ -36,6 +36,16 @@ export const every = <A>(x: Set<A>, predicate: Predicate<A>): boolean => {
   return !some(x, not(predicate))
 }
 
+/** @function */
+export const chain = <B>(bset: Setoid<B>) => <A>(x: Set<A>, f: (x: A) => Set<B>): Set<B> => {
+  let r = new Set<B>()
+  const setUnion = union(bset)
+  x.forEach(e => {
+    r = setUnion(f(e), r)
```

Comment:
I wonder, shouldn't we use mutability here for performance reasons?

Reply:
Done